client-core/src/wallet/syncer_logic.rs

```diff
@@ -221,9 +227,14 @@ fn calculate_balance_change<'a>(
                 })
             }
         }
-        Some(_spent_outputs) => {
-            // panic is impossible because total_output_ours is subset of total_output
-            let value = (total_output - total_output_ours).expect("impossible");
+        Some(spent_outputs) => {
+            let value = if transaction_type == &TransactionType::Deposit {
+                sum_outputs(spent_outputs.iter().cloned())
+                    .map_err(|_| SyncerLogicError::TotalOutputOutOfBound(encode_txid()))?
+            } else {
+                // panic is impossible because total_output_ours is subset of total_output
+                (total_output - total_output_ours).expect("impossible")
```

Comment:
impossible -> compute total_output ?

Reply:
// panic is impossible because always total_output_ours  <= subset of total_output ?

Reply:
ok